app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class CommentsController < ApplicationController
+  def new
+    @giveaway = Giveaway.find(params[:giveaway_id])
+    @comment = @giveaway.comments.build
+  end
+
+  def create
+    giveaway = Giveaway.find(params[:giveaway_id])
+    comment = giveaway.comments.build(comment_params)
+
+    comment.save
```

Comment:
I believe you need a condition here to catch behavior if comment is not saved.

Reply:
@violeta-p Yes, that is true. I will make the change. Thank you